modules/e2ee/Context.js

```diff
@@ -220,24 +302,45 @@ export class Context {
 
         if (this._cryptoKeyRing[keyIndex]) {
             const counterLength = 1 + ((data[encodedFrame.data.byteLength - 1] >> 4) & 0x7);
+            const signatureLength = data[encodedFrame.data.byteLength - 1] & 0x80
+                ? this._signatureOptions.byteLength : 0;
             const frameHeader = new Uint8Array(encodedFrame.data, 0, UNENCRYPTED_BYTES[encodedFrame.type]);
 
             // Extract the truncated authentication tag.
             const authTagOffset = encodedFrame.data.byteLength - (DIGEST_LENGTH[encodedFrame.type]
-                + counterLength + 1);
+                + counterLength + signatureLength + 1);
             const authTag = encodedFrame.data.slice(authTagOffset, authTagOffset
                 + DIGEST_LENGTH[encodedFrame.type]);
 
-            // Set authentication tag bytes to 0.
-            const zeros = new Uint8Array(DIGEST_LENGTH[encodedFrame.type]);
+            // Verify the long-term signature of the authentication tag.
+            if (signatureLength) {
+                const signature = data.subarray(data.byteLength - signatureLength - 1, data.byteLength - 1);
+                const validSignature = this._signatureKey
+                    ? await crypto.subtle.verify(this._signatureOptions,
+                        this._signatureKey, signature, authTag)
+                    : false;
```

Comment:
since restructuring the code this can no longer happen. If we receive a (key) frame with a signature but have no key yet, should we buffer? Odd timing problem but...